backend/fallback/condition_parser.py

```
@@ -0,0 +1,463 @@
+        self.current = 0
+        
+        def evaluator(context: Dict[str, Union[int, float]]) -> bool:
+            """Evaluator closure that captures parsed structure."""
+            self.current = 0
+            result = self._parse_or_expression(context)
+            
+            # Ensure all tokens consumed
+            if self._current_token().type != TokenType.EOF:
+                raise ValueError(
+                    f"Unexpected token at position {self.current}: {self._current_token()}"
+                )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)